### PR TITLE
be nicer about deleting the group lock after deleting a group

### PR DESCRIPTION
**Pull request by radix**
This implements my suggestion in https://github.com/rackerlabs/otter/issues/1192#issuecomment-89151150

I realize that the problem of deleting locks while they're still being used will be far less likely when Manish's work is done, but I think this is still a better strategy for lock cleanup when it comes to bad cases.

### CODE REVIEW

**Comment by radix** [comment]
jenkins please run integration tests

**Comment by radix** [comment]
jenkins please run integration tests

File: otter/models/cass.py [hunk 1]
```diff
@@ -1231,15 +1232,18 @@ def _delete_group():
             return d
 
         def _delete_lock_znode(result):
-            d = self.kz_client.delete(LOCK_PATH + '/' + self.uuid,
-                                      recursive=True)
```
Comment (lvh) [inline otter/models/cass.py]: Why can we get rid of the recursive here? ISTR there's nothing there, but...

File: otter/models/cass.py [hunk 2]
```diff
@@ -1231,15 +1232,18 @@ def _delete_group():
...
+                can_retry=retry_times(60),
+                next_interval=repeating_interval(5),
+                clock=self.reactor,
+            )
             d.addErrback(
                 lambda f: self.log.msg(
                     "Error cleaning up lock path (when deleting group)",
                     exc=f.value,
                     otter_msg_type="ignore-delete-lock-error"))
-            return d
```
Comment (lvh) [inline otter/models/cass.py]: The asynchrony here (well, not waiting until deleted) is intentional, right?

Comment (radix) [inline otter/models/cass.py]: Yep

**Comment by lvh** [comment]
Sorry about those gnarly mock tests. LGTM assuming my questions pan out. Thanks for working on this!

File: otter/models/cass.py (hunk 1 above)
Comment (radix) [inline otter/models/cass.py]: There _is_ stuff inside of these lock directories, and the problem is that deleting them recursively while someone else is trying to acquire the lock (or while it's held) can wreak havoc. There is content inside the lock directory only while the lock is being acquired or held.

Deleting it with recursive=False means that, if the lock is in use, the deletion will fail, which is why I added the 5 minutes of retries every 5 seconds -- this is very likely to delete the lock given our usage patterns, but will eventually give up and simply leave it there.

This PR is basically a stopgap to introducing a more persistent lock-cleanup solution based on the idea of a "child reaper" a la Curator (mentioned in the issue associated with this PR). These "child reapers" act very similar to this code, but they do it for all entries in a directory -- i.e., they list all children and constantly try to delete them non-recursively. This is pretty much the only way to definitely keep a directory full of locks clean, according to my discussion with ZK community members.

Comment (lvh) [inline otter/models/cass.py]: OK, awesome, sounds good.

**Pull request merged**